cvat/apps/engine/serializers.py

```diff
@@ -1980,63 +1982,64 @@ class Meta:
         read_only_fields = ('path',)
 
 
-def _update_related_storages(instance, validated_data):
-    for storage in ('source_storage', 'target_storage'):
-        new_conf = validated_data.pop(storage, None)
+def _update_related_storages(
+    instance: Union[models.Project, models.Task],
+    validated_data: Dict[str, Any],
+) -> None:
+    for storage_type in ('source_storage', 'target_storage'):
+        new_conf = validated_data.pop(storage_type, None)
 
         if not new_conf:
             continue
 
+        cloud_storage = None
         cloud_storage_id = new_conf.get('cloud_storage_id')
-        if cloud_storage_id:
-            _validate_existence_of_cloud_storage(cloud_storage_id)
+        new_location = new_conf.get('location')
```

Comment:
If this is called `new_location`, then the other variable ought to be called `new_cloud_storage_id`. Otherwise it's a bit unclear what's old and what's new.